drivers/junos/junos.go

```diff
@@ -0,0 +1,108 @@
+package netconf
+
+import (
+	"time"
+
+	lowlevel "github.com/arsonistgopher/go-netconf/drivers/junos/lowlevel"
+	rpc "github.com/arsonistgopher/go-netconf/rpc"
+	session "github.com/arsonistgopher/go-netconf/session"
+)
+
+// DriverJunos type is for creating a Junos based driver. Maintains state for session and connection. Implements Driver{}
+type DriverJunos struct {
+	Timeout   time.Duration    // Timeout for SSH timed sessions
+	Datastore string           // NETCONF datastore
+	Session   *session.Session // Session data
+}
+
+// New creates a new instance of DriverJunos
+func New() *DriverJunos {
+	return &DriverJunos{}
+}
+
+// SetDatastore sets the target datastore on the data structure
```

Comment:
I would not store the datastore (e.g. running or candidate) as part of the driver as this is information corresponds to the RPC. Comparing the candidate and the running configuration requires two RPC and with this approach you would like to do a `SetDatastore` before each of them.